lib/aussieaddonscommon/utils.py

```diff
@@ -218,18 +218,49 @@ def log_kodi_platform_version():
     log("Kodi %s running on %s" % (version, platform))
 
 
+
+def check_country(connection_info, message=None):
+    if not message:
+        message = format_dialog_message('Issue report denied.')
+
+    import issue_reporter
+    valid_country = issue_reporter.valid_country(connection_info)
+    blacklisted_hostname = issue_reporter.blacklisted_hostname(connection_info)
+
+    if not valid_country:
+        country_code = connection_info.get('country')
+        if country_code:
+            import countries
+            country_name = countries.countries.get(country_code, country_code)
+            message.append('Your country is reported as %s, but this service '
+                           'is probably geo-blocked to Australia.' %
+                           country_name)
+            xbmcgui.Dialog().ok(*message)
+    
+    if blacklisted_hostname:
+        message.append('VPN/proxy detected that has been blocked by this '
+                       'content provider.')
+        xbmcgui.Dialog().ok(*message)
+
+    return valid_country and not blacklisted_hostname
```

Comment:
maybe say return True here and add a return False after the blacklisted_hostname check?